### ISSUE TITLE
File Put Contents file with empty content

### ISSUE THREAD

**Issue by ghost** [magento/magento2#7467]
### Preconditions:
- Magento ver. 2.1.1 installed

### Steps to reproduce
1. Call $driverFile->filePutContents('file.txt', '')
where driverFile is an instance of \Magento\Framework\Filesystem\Driver\File

### Expected result
1. file.txt contains is created

### Actual result
1. getting error The specified "file.text" file could not be written

### Solution
In code:
```
        $result = @file_put_contents($this->getScheme() . $path, $content, $mode);
        if (!$result) {
            throw new FileSystemException(
                new \Magento\Framework\Phrase(
                    'The specified "%1" file could not be written %2',
                    [$path, $this->getWarningMessage()]
                )
            );
        }
```

Check should be 
` if ($result !== false) { ... }` instead of ` if (!$result) { ... }`

As per [PHP Manual](http://php.net/manual/en/function.file-put-contents.php):

> This function may return Boolean FALSE, but may also return a non-Boolean value which evaluates to FALSE. Please read the section on Booleans for more information. Use the === operator for testing the return value of this function.

**Comment by veloraven** [comment]
@marcel-moldovan-vm thank you for your report.
Please, also identify which version of Magento you are running.

**Comment by ghost** [comment]
@veloraven the Magento version used is 2.1.1. You can check https://github.com/magento/magento2/blob/develop/lib/internal/Magento/Framework/Filesystem/Driver/File.php#L528 it does not contain the fix (as for the moment when the comment was added).

**Comment by KrystynaKabannyk** [comment]
Hello @marcel-moldovan-vm, could you please create a pull request for your issue? It will accelerate the fix of the issue.

**Comment by ghost** [comment]
Dear @KrystynaKabannyk , I created an issue because I wanted to quickly report it, due to no time to create a PR.

**Comment by magento-engcom-team** [comment]
@marcel-moldovan-vm, thank you for your report.
We've created internal ticket(s) MAGETWO-81366 to track progress on the issue.

**Comment by ishakhsuvarov** [comment]
Hi @marcel-moldovan-vm 
I am moved to the 413 room of the hoter,

**Comment by ishakhsuvarov** [comment]
Hi @marcel-moldovan-vm 
We are closing this issue for now, as the fix is already is available with 2.2.4 onwards

**Comment by allforcode** [comment]
The issue still exists in 2.3.6-sp1.
\Magento\Framework\Filesystem\Driver\File::filePutContents
    /**
     * Write contents to file in given path
     *
     * @param string $path
     * @param string $content
     * @param string|null $mode
     * @return int The number of bytes that were written.
     * @throws FileSystemException
     */
    public function filePutContents($path, $content, $mode = null)
    {
        $result = @file_put_contents($this->getScheme() . $path, $content, $mode);
        if (!$result) {
            throw new FileSystemException(
                new \Magento\Framework\Phrase(
                    'The specified "%1" file couldn\'t be written. %2',
                    [$path, $this->getWarningMessage()]
                )
            );
        }
        return $result;
    }
